node/src/blockchain/blockchain_interface/blockchain_interface_web3/lower_level_interface_web3.rs

```diff
@@ -1,395 +1,762 @@
-        let (_event_loop_handle, transport) = Http::with_max_parallel(
-            &format!("http://{}:{}", &Ipv4Addr::LOCALHOST.to_string(), port),
-            REQUESTS_IN_PARALLEL,
-        )
-        .unwrap();
-        let chain = TEST_DEFAULT_CHAIN;
-        let subject = make_subject(transport, chain);
+    fn get_block_number_works() {
+        let port = find_free_port();
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .response("0x23".to_string(), 1)
+            .start();
+        let subject = make_blockchain_interface_web3(Some(port));
 
-        let result = subject.get_transaction_fee_balance(&Wallet::new("0x_invalid_wallet_address"));
+        let result = subject.lower_interface().get_block_number().wait();
 
-        assert_eq!(result, Err(BlockchainError::InvalidAddress));
+        assert_eq!(result, Ok(35.into()));
     }
 
     #[test]
-    fn low_interface_web3_get_transaction_fee_balance_returns_err_for_unintelligible_response() {
-        let act = |subject: &LowBlockchainIntWeb3<Http>, wallet: &Wallet| {
-            subject.get_transaction_fee_balance(wallet)
-        };
+    fn get_block_number_returns_an_error() {
+        let port = find_free_port();
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .response("trash".to_string(), 1)
+            .start();
+        let subject = make_blockchain_interface_web3(Some(port));
+
+        let error = subject
+            .lower_interface()
+            .get_block_number()
+            .wait()
+            .unwrap_err();
 
-        assert_error_from_unintelligible_response(act, "invalid hex character");
+        assert_eq!(
+            error,
+            QueryFailed(
+                "Decoder error: Error(\"0x prefix is missing\", line: 0, column: 0)".to_string()
+            )
+        );
     }
 
     #[test]
-    fn low_interface_web3_get_masq_balance_works() {
+    fn get_transaction_id_works() {
         let port = find_free_port();
-        let test_server = TestServer::start (port, vec![
-            br#"{"jsonrpc":"2.0","id":0,"result":"0x00000000000000000000000000000000000000000000000000000000DEADBEEF"}"#.to_vec()
-        ]);
-        let (_event_loop_handle, transport) = Http::with_max_parallel(
-            &format!("http://{}:{}", &Ipv4Addr::LOCALHOST.to_string(), port),
-            REQUESTS_IN_PARALLEL,
-        )
-        .unwrap();
-        let chain = TEST_DEFAULT_CHAIN;
-        let subject = make_subject(transport, chain);
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .response("0x23".to_string(), 1)
+            .start();
+        let subject = make_blockchain_interface_web3(Some(port));
+        let wallet = &Wallet::from_str("0x3f69f9efd4f2592fd70be8c32ecd9dce71c472fc").unwrap();
+
+        let result = subject
+            .lower_interface()
+            .get_transaction_id(wallet.address())
+            .wait();
+
+        assert_eq!(result, Ok(35.into()));
```

Comment:
Same problem as previously (the returned number assertion and some assertions for the direction out).